app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use App\Models\Category;
+use App\Models\User;
+use App\Models\Image;
+use Illuminate\Http\Request;
+use App\Http\Requests\PostRequest;
+use Illuminate\Support\Facades\DB;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $posts = Post::all();
+
+        return view('Admin/index', compact(['posts']));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::all();
+
+        return view('Admin/create', compact(['categories']));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(PostRequest $request)
+    {
+        $post = new Post;
+        $email = $request->email;
+        $user_id = User::where('email', 'like', $email)->first();
+        $category = $request->category;
+        $category_id = Category::where('id', 'like', $category)->first();
+        
+        $post->title = $request->title;
+        $post->body = $request->body;
+        $post->author_id = $user_id->id;
+        $post->category_id = $category_id->id;
+        $post->save();
+
+        $image = $request->image;
+        $image_post = new Image;
+        $image_post->imageable_type = 'Post';
+        $image_post->imageable_id = $post->id;
+        $image_post->link = $image;
+        $image_post->save();
+
+        return redirect('admin');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\Post  $post
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+        $post = Post::findOrFail($id);
+        $image_post = Post::findOrFail($id)->load('images');
+
+        return view('Admin/detail', compact(['post', 'image_post']));
```

Comment:
```suggestion
        $post = Post::findOrFail($id)->load('images');

        return view('admin.detail', compact(['post']));
```
Sửa tương tự các chỗ khác nhé :3 m find 2 lần làm chi :3 find 1 lần là đc rùi.